src/auto/assistant.rs

```diff
@@ -233,9 +234,27 @@ impl<O: IsA<Assistant>> AssistantExt for O {
         }
     }
 
-    //fn set_forward_page_func<'a, P: Into<Option<&'a /*Unimplemented*/AssistantPageFunc>>, Q: Into<Option</*Unimplemented*/Fundamental: Pointer>>>(&self, page_func: P, data: Q, destroy: /*Unknown conversion*//*Unimplemented*/DestroyNotify) {
-    //    unsafe { TODO: call ffi::gtk_assistant_set_forward_page_func() }
-    //}
+    fn set_forward_page_func<P: Fn(i32) -> i32 + 'static, Q: Into<Option<P>>>(&self, page_func: Q) {
+        let page_func = page_func.into();
+        let page_func_data: Box_<Box_<Option<P>>> = Box::new(Box::new(page_func.into()));
```

Comment:
As callback and destroy notify are working on the concrete type `P`, you only need to box once here. Boxing twice is only needed for trait objects